public/components/management/groups/multiple-agent-selector.tsx

```diff
@@ -9,707 +9,897 @@
+              'r'
+            );
+          }
+        });
+
+        this.checkLimit();
+
+        //relaod agent columns
+        this.reload('right');
+        this.reload('left');
+
+        try {
+          //delete nodes selected
+          await WzRequest.apiReq('DELETE', `/agents/group`, {
+            params: {
+              group_id: this.props.currentGroup.name,
+              agents_list: itemsToSave.addedIds.toString(),
+            },
+          });
+        } catch (e) {
+          console.error(e);
+        }
+
+        const options = {
+          context: `${MultipleAgentSelector.name}.saveAddAgents`,
+          level: UI_LOGGER_LEVELS.ERROR,
+          severity: UI_ERROR_SEVERITIES.BUSINESS,
+          error: {
+            error: {
+              message: error,
+              stack: error,
+            },
+            message: error.message || error,
+            title: error.name,
+          },
+        };
+
+        getErrorOrchestrator().handleError(options);
       }
-      this.setState({ savingChanges: false });
-      this.props.cancelButton();
-    } catch (error) {
-      this.setState({ savingChanges: false });
-      const options = {
-        context: `${MultipleAgentSelector.name}.saveAddAgents`,
-        level: UI_LOGGER_LEVELS.ERROR,
-        severity: UI_ERROR_SEVERITIES.BUSINESS,
-        error: {
-          error: error,
-          message: error.message || error,
-          title: `${error.name}: Error applying changes`,
-        },
-      };
-      getErrorOrchestrator().handleError(options);
+      return;
     }
-    return;
-  }
 
-  clearFailedErrors() {
-    this.failedErrors = false;
-  }
+    clearFailedErrors() {
+      this.failedErrors = false;
+    }
 
-  checkLimit() {
-    if (this.firstSelectedList) {
-      const itemsToSave = this.getItemsToSave();
-      const currentAdding = itemsToSave.addedIds.length;
-      const currentDeleting = itemsToSave.deletedIds.length;
-      this.setState({
-        currentAdding,
-        currentDeleting,
-        moreThan500: currentAdding > 500 || currentDeleting > 500
-      })
+    checkLimit() {
+      if (this.firstSelectedList) {
+        const itemsToSave = this.getItemsToSave();
+        const currentAdding = itemsToSave.addedIds.length;
+        const currentDeleting = itemsToSave.deletedIds.length;
+        this.setState({
+          currentAdding,
+          currentDeleting,
+          moreThan500: currentAdding > 500 || currentDeleting > 500,
+        });
+      }
     }
-  }
 
-  moveItem = (item, from, to, type) => {
-    if (Array.isArray(item)) {
-      item.forEach(elem => this.moveItem(elem, from, to, type));
-      this.checkLimit();
-    } else {
-      item = JSON.parse(item);
-      const idx = from.findIndex(x => x.key === item.key);
-      if (idx !== -1) {
-        from.splice(idx, 1);
-        item.type = !item.type ? type : '';
-        to.push(item)
+    moveItem = (item, from, to, type) => {
```

Comment:
This functions mutates the state. It doesn't seem to have been changed in this PR but it should be addressed